From 45788eee6f9fb7909808b7c21665ee2f7d2faa96 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Sat, 28 Dec 2019 02:13:55 +0100
Subject: [PATCH 074/122] ARM: dts: mxiii: add support for the audio outputs

The MXIII and MXIII-Plus boards come with SPDIF and I2S Stereo DAC
(provided by a Beijing Times Minxin Technology Co., Ltd. MXT8234) audio
outputs. The latter is not supported yet (as we are missing a driver for
it). Also the I2S output can be routed to the HDMI TX controller.

Add the SPDIF output as well as HDMI TX audio routing on these boards.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 arch/arm/boot/dts/meson8m2-m8s.dts    |  5 ---
 arch/arm/boot/dts/meson8m2-mxiii.dtsi | 62 +++++++++++++++++++++++++++
 2 files changed, 62 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/meson8m2-m8s.dts b/arch/arm/boot/dts/meson8m2-m8s.dts
index d2210a983955..d814b160943c 100644
--- a/arch/arm/boot/dts/meson8m2-m8s.dts
+++ b/arch/arm/boot/dts/meson8m2-m8s.dts
@@ -86,8 +86,6 @@ sound {
 		assigned-clocks = <&clkc CLKID_MPLL0>,
 				  <&clkc CLKID_MPLL1>,
 				  <&clkc CLKID_MPLL2>;
-		assigned-clock-rates = <270950400>,
-				       <294912000>,
 		assigned-clock-rates = <294912000>,
 				       <270950400>,
 				       <393216000>;
@@ -205,14 +203,11 @@ xtal_32k_out: xtal-32k-out-clk {
 
 &aiu {
 	status = "okay";
-<<<<<<< HEAD
-=======
 
 	pinctrl-0 = <&i2s_am_clk_pins>, <&i2s_out_ao_clk_pins>,
 		    <&i2s_out_lr_clk_pins>, <&i2s_out_ch01_ao_pins>,
 		    <&spdif_out_pins>;
 	pinctrl-names = "default";
->>>>>>> 7e69df433bae (ARM: dts: meson8m2: m8s: enable the audio outputs - WiP)
 };
 
 &cec_AO {
diff --git a/arch/arm/boot/dts/meson8m2-mxiii.dtsi b/arch/arm/boot/dts/meson8m2-mxiii.dtsi
index b6f4e0e1274a..bc13c8f7ebf8 100644
--- a/arch/arm/boot/dts/meson8m2-mxiii.dtsi
+++ b/arch/arm/boot/dts/meson8m2-mxiii.dtsi
@@ -64,6 +64,59 @@ sdio_pwrseq: sdio-pwrseq {
 		clock-names = "ext_clock";
 	};
 
+	sound {
+		compatible = "amlogic,gx-sound-card";
+		model = "M8M2-MXIII";
+		audio-routing = "I2S Encoder Playback", "I2S FIFO Playback",
+				"SPDIF Encoder Playback", "I2S FIFO Playback",
+				"SPDIF Encoder Playback", "SPDIF FIFO Playback";
+
+		assigned-clocks = <&clkc CLKID_MPLL0>,
+				  <&clkc CLKID_MPLL1>;
+		assigned-clock-rates = <270950400>,
+				       <294912000>;
+
+		dai-link-0 {
+			sound-dai = <&aiu AIU_CPU CPU_I2S_FIFO>;
+		};
+
+		dai-link-1 {
+			sound-dai = <&aiu AIU_CPU CPU_SPDIF_FIFO>;
+		};
+
+		dai-link-2 {
+			sound-dai = <&aiu AIU_CPU CPU_I2S_ENCODER>;
+			dai-format = "i2s";
+			mclk-fs = <256>;
+
+			codec-0 {
+				sound-dai = <&aiu AIU_HDMI CTRL_I2S>;
+			};
+		};
+
+		dai-link-3 {
+			sound-dai = <&aiu AIU_HDMI CTRL_OUT>;
+
+			codec-0 {
+				sound-dai = <&hdmi_tx 0>;
+			};
+		};
+
+		dai-link-4 {
+			sound-dai = <&aiu AIU_CPU CPU_SPDIF_ENCODER>;
+
+			codec-0 {
+				sound-dai = <&spdif_dit>;
+			};
+		};
+	};
+
+	spdif_dit: spdif-audio-codec {
+		compatible = "linux,spdif-dit";
+		#sound-dai-cells = <0>;
+		sound-name-prefix = "DIT";
+	};
+
 	cvbs-connector {
 		compatible = "composite-video-connector";
 
@@ -89,6 +142,15 @@ xtal_32k_out: xtal-32k-out-clk {
 	};
 };
 
+&aiu {
+	status = "okay";
+
+	pinctrl-0 = <&i2s_am_clk_pins>, <&i2s_out_ao_clk_pins>,
+		    <&i2s_out_lr_clk_pins>, <&i2s_out_ch01_ao_pins>,
+		    <&spdif_out_pins>;
+	pinctrl-names = "default";
+};
+
 &cpu0 {
 	cpu-supply = <&vcck>;
 };
-- 
2.25.1

